client/viame-web-common/components/AttributeInput.vue

```diff
@@ -19,58 +24,146 @@ export default Vue.extend({
       type: Array,
       default: () => [],
     },
+    focus: {
+      type: Boolean,
+      default: false,
+    },
   },
+  setup(props, { emit }) {
+    const tempVal = ref(null as null | boolean | number | string);
+    const inputBoxRef = ref(undefined as undefined | HTMLInputElement);
+    const boolOpts = [
+      { text: '', value: undefined },
+      { text: 'true', value: true },
+      { text: 'false', value: false },
+    ];
+    tempVal.value = props.value;
+    onMounted(() => {
+      if (props.focus && inputBoxRef.value) {
+        inputBoxRef.value.focus();
+      }
+    });
 
-  computed: {
-    foo() { return this.value; },
-  },
-
-  methods: {
-    change(newval: string | boolean | number | undefined): void {
-      const { name } = this;
-      let value;
-      switch (newval) {
-        case '':
-        case undefined:
-          value = undefined;
+    function blurType(e: KeyboardEvent) {
+      (e.target as HTMLInputElement).blur();
+    }
+    function onFocus() {
+      if (props.values && props.values.length) {
+        tempVal.value = null;
```

Comment:
Potentially prefer to just select-all the existing text here.

`inputBoxRef.value.select()` should do it.

Reply:
The Datalist thing is annoying in that it won't display the dropdown unless the current value is null for the input element.  So if you have any default `props.values` we need to null it out so you can pick from the list or type in your own.  I'll update it so that if there are no default values (`props.values.length === 0`) it will select the current text to allow typing.  I double checked and this seems to be the same behavior between the type dropdowns in the tracklist and on the MDN site.